### DISCUSSION TITLE
How to read the "Discover" Feed?

### DISCUSSION THREAD

**Discussion by MatthiasThoemel** [MarshalX/atproto#407] [Questions] [answered]
Hello,

I am trying to read the Discover fead for a bot functionality. If you do it by hand in a browser, it is this URL:

https://bsky.app/profile/did:plc:z72i7hdynmk6r22z27h6tvur/feed/whats-hot

But I was not able to do this by atproto. 

Basically, the Discover feed is a (dynamic) feed generated by the bsky.app user. But atproto complains about the AT proto syntax when I try to read this by get_posts(...). Or, if it does not complain the bsky server answers with a 502 error. 

Can somebody help me in reading the Discover feed ?

My best regards, Matthias

**Comment by MarshalX** [comment, marked as answer]
Hello,

```python
from atproto import Client, models


def main() -> None:
    client = Client()
    client.login('handle', 'password')

    discover_feed_uri = 'at://did:plc:z72i7hdynmk6r22z27h6tvur/app.bsky.feed.generator/whats-hot'

    feed_response = client.app.bsky.feed.get_feed(models.AppBskyFeedGetFeed.Params(feed=discover_feed_uri))
    for post_view in feed_response.feed:
        author = post_view.post.author.handle
        text = post_view.post.record.text.replace('\n', ' ')
        print(f'{author}: {text}')
```

Keep in mind that you need to implement pagination using `feed_response.cursor` if you want to fetch more posts. Here is the [example](https://github.com/MarshalX/atproto/blob/main/examples/advanced_usage/handle_cursor_pagination.py)

**Reply by MatthiasThoemel** [reply]
thank you very much. I have another question :-) "How to repost quoted to a post in a feed #408"
